### PR TITLE
Adding computation

### PR DESCRIPTION
**Pull request by gavinbell123**

### CODE REVIEW

**Review by Mario64iscool2 — CHANGES_REQUESTED** [review]
`compute` as a method is supposed to be the sole outward-facing method, Our interfaces do not contain methods that only the computation class cares to know about. Anything interacting with an `ICompute` only needs to know that it can compute the result of an integer passed to it. Interfaces are only for methods that interacting classes need to know about.

**Review by Mario64iscool2 — CHANGES_REQUESTED** [review]
Make these changes, and submit them in a single commit.

File: src/compute/Engine.java [hunk 1]
```diff
@@ -31,5 +31,17 @@ public String compute(InputSource<?> src, OutputSource dst) {
 		
 		return "".toString();
 	}
+	@Override
```
Comment (Mario64iscool2) [inline src/compute/Engine.java:34]:
```suggestion

```

File: src/compute/ICompute.java [hunk 2]
```diff
@@ -12,5 +12,8 @@ public interface ICompute {
 	 * @return the resultant long
 	 */
 	long compute(int i);
+
+	boolean isxPrime(int x);
+	int firstgreaterprimeNum(int x, int y);
```
Comment (Mario64iscool2) [inline src/compute/ICompute.java:17]:
```suggestion

```
These methods are only pertinent to our specific implementation of a Compute Engine. They should not be in the interface. Interfaces are only for methods that interacting classes should know about. In our case, the only method that a user of an `ICompute`-implementing object needs to know about is the `compute(int i)` method. A caller of this method trusts that the implementing class will do as it is told. It doesn't need to know about the inner workings of the class.

**Review by Mario64iscool2 — CHANGES_REQUESTED** [review]

**Review by Melataylor519 — APPROVED** [review]
Looks good

**Comment by Mario64iscool2** [comment]
@gavinbell123 If your stuff doesn't compile locally, please fix that before pushing to a repository. If you don't know whether or not something will compile, a development environment like [Eclipse](https://www.eclipse.org/) will be your best friend.
Last warning.

**Pull request merged**